stdlib/public/core/StringCreate.swift

```diff
@@ -13,60 +13,126 @@
+    // initializer directly, though.)
+    //
+    // ¹ well, most of the time, which makes it a rather conditional
+    // "unconditionally".
+    return unsafe input.allSatisfy { $0 < 0x80 }
+  }
+  
+  // bytes.count is non-zero, so we can unconditionally unwrap baseAddress.
+  let base = unsafe input.baseAddress._unsafelyUnwrappedUnchecked
+  let n = input.count
+  var i = 0
   
-  if unsafe ptr < end16 {
-    let value = unsafe ptr.loadUnaligned(as: UInt16.self)
-    guard value & asciiMask16 == 0 else { return false }
-    unsafe ptr = unsafe ptr + MemoryLayout<UInt16>.stride
+  guard n >= MemoryLayout<Block>.size else {
+    // The size isn't yet to a block
+    // word-by-word we are forced to walk.
+    // So as to not leave a gap
+    // the last word may lap
+    // the word that we already chalked.
+    //
+    //     0      k      2k     3k      ?k   n-k    n-1
+    //     |      |      |      |       |     |      |
+    //     +------+------+------+       +------+     |
+    //     | word | word | word |  ...  | word |     |
+    //     +------+------+------+       +------+     v
+    //                                        +------+
+    //      possibly overlapping final word > | word |
+    //                                        +------+
+    //
+    // This means that we check any bytes in the overlap region twice, but
+    // that's much preferrable to using smaller accesses to avoid rechecking,
+    // because the entire last word is about as expensive as checking just
+    // one byte would be, and on average there's more than one byte remaining.
+    //
+    // Note that we don't bother trying to align any of these accesses, because
+    // there is minimal benefit to doing so on "modern" OoO cores, which can
+    // handle cacheline-crossing loads at full speed. If the string happens to
+    // be aligned, they'll be aligned, if not, they won't be. It will likely
+    // make sense to add a path that does align everything for more limited
+    // embedded CPUs, though.
+    let k = MemoryLayout<Word>.size
+    let last = n &- k
+    while i < last {
+      guard unsafe allASCII(wordAt: base + i) else { return false }
+      i &+= k
+    }
+    return unsafe allASCII(wordAt: base + last)
   }
-
-  if unsafe ptr < end {
-    let value = unsafe ptr.loadUnaligned(fromByteOffset: 0, as: UInt8.self)
-    guard value & asciiMask8 == 0 else { return false }
+  
+  // check block-by-block, with a possibly overlapping last block to avoid
+  // sub-block cleanup. We should be able to avoid manual index arithmetic
+  // and write this loop and the one above something like the following:
+  //
+  //  return stride(from: 0, to: last, by: k).allSatisfy {
+  //    allASCII(blockAt: base + $0)
+  //  } && allASCII(blockAt: base + last)
+  //
+  // but LLVM leaves one unnecessary conditional operation in the loop
+  // when we do that, so we write them out as while loops instead for now.
```

Comment:
I feel like I'm repeating your own words to yourself, but we might want to link to an llvm issue/enhancement here.